<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>POMA Workflow Overview • POMA</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/spacelab/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="POMA Workflow Overview">
<meta property="og:description" content="POMA">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">POMA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.18</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/POMA-demo.html">`POMA` Demo
</a>
    </li>
    <li>
      <a href="../articles/POMA-normalization.html">`POMA` Normalization
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/pcastellanoescuder/POMA">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>POMA Workflow Overview</h1>
                        <h4 class="author">Pol Castellano-Escuder</h4>
            <address class="author_afil">
      University of Barcelona, Spain.<br><a class="author_email" href="mailto:#"></a><a href="mailto:polcaes@gmail.com" class="email">polcaes@gmail.com</a>
      </address>
                  
            <h4 class="date">2020-06-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/pcastellanoescuder/POMA/blob/master/vignettes/POMA-demo.Rmd"><code>vignettes/POMA-demo.Rmd</code></a></small>
      <div class="hidden name"><code>POMA-demo.Rmd</code></div>

    </div>

    
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Run the following code to install the Bioconductor version of package. <strong><code>POMA</code> is not yet in Bioconductor! Hope very soon…</strong></p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="co"># install.packages("BiocManager")</span>
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"POMA"</span>)</pre></body></html></div>
<p>Alternatively, you can install the latest development version of <code>POMA</code> from Github.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co"># install.packages("devtools")</span>
<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"pcastellanoescuder/POMA"</span>)</pre></body></html></div>
</div>
<div id="load-poma" class="section level1">
<h1 class="hasAnchor">
<a href="#load-poma" class="anchor"></a>Load POMA</h1>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">POMA</span>)</pre></body></html></div>
<p>You can also load some additional packages that will be very useful in this vignette.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggraph</span>)</pre></body></html></div>
</div>
<div id="the-poma-workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#the-poma-workflow" class="anchor"></a>The POMA Workflow</h1>
<p><code>POMA</code> functions can be divided in three sequential well separated blocks: <strong>Data Preparation</strong>, <strong>Pre-processing</strong> and <strong>Statistical Analysis</strong>.</p>
<div id="data-preparation" class="section level2">
<h2 class="hasAnchor">
<a href="#data-preparation" class="anchor"></a>Data Preparation</h2>
<p>The <strong>MSnbase</strong> Bioconductor package provides a well defined computational data structures to represent mass spectrometry (MS) experiment data types <span class="citation">(Gatto and Lilley <a href="#ref-MSnbase">2012</a>)</span><span class="citation">(Huber et al. <a href="#ref-Biobase">2015</a>)</span>. Since data structures can mean a marked improvement in data analysis, <strong>POMA</strong> functions use <strong>MSnSet</strong> objects from <strong>MSnbase</strong> package, allowing the reusability of existing methods for this class and contributing to the improvement of robust and reproducible workflows.</p>
<p>The first step of workflow will be load or create an <code><a href="https://rdrr.io/pkg/MSnbase/man/MSnSet-class.html">MSnbase::MSnSet()</a></code> object. Often, you will have your data stored in separated matrix and/or data frames and you will have to create your own <code>MSnSet</code> object. <code>PomaMSnSetClass</code> function makes this step fast and easy building this <code>MSnSet</code> object from your independent files.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="co"># create an MSnSet object from two separated data frames</span>
<span class="no">target</span> <span class="kw">&lt;-</span> <span class="kw pkg">readr</span><span class="kw ns">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_csv</a></span>(<span class="st">"your_target.csv"</span>)
<span class="no">features</span> <span class="kw">&lt;-</span> <span class="kw pkg">readr</span><span class="kw ns">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_csv</a></span>(<span class="st">"your_features.csv"</span>)

<span class="no">data</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/PomaMSnSetClass.html">PomaMSnSetClass</a></span>(<span class="kw">target</span> <span class="kw">=</span> <span class="no">target</span>, <span class="kw">features</span> <span class="kw">=</span> <span class="no">features</span>)</pre></body></html></div>
<p>Alternatively, if your data is already stored in a <code>MSnSet</code> object, you can skip this step and go directly to the Pre-processing step. In this vignette we will use the example data provided in the package.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"># load example data
data("st000336")
st000336
&gt; MSnSet (storageMode: lockedEnvironment)
&gt; assayData: 31 features, 57 samples 
&gt;   element names: exprs 
&gt; protocolData: none
&gt; phenoData
&gt;   sampleNames: DMD004.1.U02 DMD005.1.U02 ... DMD173.1.U02 (57 total)
&gt;   varLabels: group steroids
&gt;   varMetadata: labelDescription
&gt; featureData: none
&gt; experimentData: use 'experimentData(object)'
&gt; Annotation:  
&gt; - - - Processing information - - -
&gt;  MSnbase version: 2.12.0</pre></body></html></div>
<div id="brief-description-of-example-data" class="section level3">
<h3 class="hasAnchor">
<a href="#brief-description-of-example-data" class="anchor"></a>Brief Description of Example Data</h3>
<p>This example data is composed of 57 samples, 31 metabolites, 1 covariate and 2 experimental groups (Controls and DMD) from a targeted LC/MS study.</p>
<p><em>Duchenne Muscular Dystrophy (DMD) is an X-linked recessive form of muscular dystrophy that affects males via a mutation in the gene for the muscle protein, dystrophin. Progression of the disease results in severe muscle loss, ultimately leading to paralysis and death. Steroid therapy has been a commonly employed method for reducing the severity of symptoms. This study aims to quantify the urine levels of amino acids and organic acids in patients with DMD both with and without steroid treatment. Track the progression of DMD in patients who have provided multiple urine samples.</em></p>
<p>This data was collected from <a href="https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&amp;DataMode=AllData&amp;StudyID=ST000336&amp;StudyType=MS&amp;ResultType=1#DataTabs">here</a>.</p>
</div>
</div>
<div id="pre-processing" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-processing" class="anchor"></a>Pre Processing</h2>
<p>This is a critical point in the workflow because all final statistical results will depend on the decisions made here. Again, this block can be divided in 3 steps: <strong>Missing Value Imputation</strong>, <strong>Normalization</strong> and <strong>Outlier Detection</strong>.</p>
<div id="missing-value-imputation" class="section level3">
<h3 class="hasAnchor">
<a href="#missing-value-imputation" class="anchor"></a>Missing Value Imputation</h3>
<p>Often, due to biological and technical reasons, some features can not be identified or quantified in some samples in MS <span class="citation">(Armitage et al. <a href="#ref-imputation">2015</a>)</span>. <strong>POMA</strong> offers 7 different imputation methods to deal with this situation. Just run the following line of code to impute your missings!</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r">imputed &lt;- PomaImpute(st000336, ZerosAsNA = TRUE, RemoveNA = TRUE, cutoff = 20, method = "knn")
imputed
&gt; MSnSet (storageMode: lockedEnvironment)
&gt; assayData: 30 features, 57 samples 
&gt;   element names: exprs 
&gt; protocolData: none
&gt; phenoData
&gt;   sampleNames: DMD004.1.U02 DMD005.1.U02 ... DMD173.1.U02 (57 total)
&gt;   varLabels: Group steroids
&gt;   varMetadata: labelDescription
&gt; featureData: none
&gt; experimentData: use 'experimentData(object)'
&gt; Annotation:  
&gt; - - - Processing information - - -
&gt; Imputed (knn): Fri Jun 26 14:19:09 2020 
&gt;  MSnbase version: 2.14.2</pre></body></html></div>
<p><em>Note that the object has been updated with imputation information.</em></p>
</div>
<div id="normalization" class="section level3">
<h3 class="hasAnchor">
<a href="#normalization" class="anchor"></a>Normalization</h3>
<p>The next step of this block is the data normalization. Often, some factors can introduce variability in some types of MS data having a critical influence on the final statistical results, making normalization a key step in the workflow <span class="citation">(Berg et al. <a href="#ref-normalization">2006</a>)</span>. Again, <strong>POMA</strong> offers several methods to normalize the data by running just the following line of code:</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r">normalized &lt;- PomaNorm(imputed, method = "log_pareto")
normalized
&gt; MSnSet (storageMode: lockedEnvironment)
&gt; assayData: 30 features, 57 samples 
&gt;   element names: exprs 
&gt; protocolData: none
&gt; phenoData
&gt;   sampleNames: DMD004.1.U02 DMD005.1.U02 ... DMD173.1.U02 (57 total)
&gt;   varLabels: Group steroids
&gt;   varMetadata: labelDescription
&gt; featureData: none
&gt; experimentData: use 'experimentData(object)'
&gt; Annotation:  
&gt; - - - Processing information - - -
&gt; Imputed (knn): Fri Jun 26 14:19:09 2020 
&gt; Normalised (log_pareto): Fri Jun 26 14:19:09 2020 
&gt;  MSnbase version: 2.14.2</pre></body></html></div>
<p><em>Note that the object has been updated with normalization information.</em></p>
<div id="normalization-effect" class="section level4">
<h4 class="hasAnchor">
<a href="#normalization-effect" class="anchor"></a>Normalization effect</h4>
<p>Sometimes, you will be interested in <em>how the normalization process affect your data</em>?</p>
<p>To answer this question, <strong>POMA</strong> offers two exploratory functions, <code>PomaBoxplots</code> and <code>PomaDensity</code>, that can help to understand the normalization process.</p>
<p><code>PomaBoxplots</code> generates boxplots for all samples or features (depending on the group factor) of an <code>MSnSet</code> object. Here, we can compare objects before and after normalization step.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="../reference/PomaBoxplots.html">PomaBoxplots</a></span>(<span class="no">imputed</span>, <span class="kw">group</span> <span class="kw">=</span> <span class="st">"samples"</span>, <span class="kw">jitter</span> <span class="kw">=</span> <span class="fl">FALSE</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Not Normalized"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">legend.position</span> <span class="kw">=</span> <span class="st">"none"</span>) <span class="co"># data before normalization</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-10-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="../reference/PomaBoxplots.html">PomaBoxplots</a></span>(<span class="no">normalized</span>, <span class="kw">group</span> <span class="kw">=</span> <span class="st">"samples"</span>, <span class="kw">jitter</span> <span class="kw">=</span> <span class="fl">FALSE</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Normalized"</span>) <span class="co"># data after normalization</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-11-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<p>On the other hand, <code>PomaDensity</code> shows the distribution of all features before and after the normalization process.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="fu"><a href="../reference/PomaDensity.html">PomaDensity</a></span>(<span class="no">imputed</span>, <span class="kw">group</span> <span class="kw">=</span> <span class="st">"features"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Not Normalized"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">legend.position</span> <span class="kw">=</span> <span class="st">"none"</span>) <span class="co"># data before normalization</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-12-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="../reference/PomaDensity.html">PomaDensity</a></span>(<span class="no">normalized</span>, <span class="kw">group</span> <span class="kw">=</span> <span class="st">"features"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Normalized"</span>) <span class="co"># data after normalization</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-13-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="outlier-detection" class="section level3">
<h3 class="hasAnchor">
<a href="#outlier-detection" class="anchor"></a>Outlier Detection</h3>
<p>Finally, the last step of this block is the Outlier Detection. Outlers are defined as observations that are not concordant with those of the vast majority of the remaining data points. These values can have an enormous influence on the resultant statistical analysis, being a dangerous ground for all required assumptions in the most commonly applied parametric tests in mass spectrometry as well as for all also required assumptions in many regression techniques and predictive modeling approaches. <strong>POMA</strong> allows the analysis of outliers as well as the possibility to remove them from the analysis using different modulable parameters.</p>
<p>Analyze and remove outliers running the following two lines of code.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="../reference/PomaOutliers.html">PomaOutliers</a></span>(<span class="no">normalized</span>, <span class="kw">do</span> <span class="kw">=</span> <span class="st">"analyze"</span>)$<span class="no">polygon_plot</span> <span class="co"># to explore</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-14-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb14"><html><body><pre class="r">pre_processed &lt;- PomaOutliers(normalized, do = "clean") # to remove outliers
pre_processed
&gt; MSnSet (storageMode: lockedEnvironment)
&gt; assayData: 30 features, 50 samples 
&gt;   element names: exprs 
&gt; protocolData: none
&gt; phenoData
&gt;   sampleNames: DMD004.1.U02 DMD005.1.U02 ... DMD173.1.U02 (50 total)
&gt;   varLabels: Group steroids
&gt;   varMetadata: labelDescription
&gt; featureData: none
&gt; experimentData: use 'experimentData(object)'
&gt; Annotation:  
&gt; - - - Processing information - - -
&gt; Imputed (knn): Fri Jun 26 14:19:09 2020 
&gt; Normalised (log_pareto): Fri Jun 26 14:19:09 2020 
&gt; Outliers removed (euclidean and median): Fri Jun 26 14:19:16 2020 
&gt;  MSnbase version: 2.14.2</pre></body></html></div>
<p><em>Note that the object has been updated with outlier information.</em></p>
</div>
</div>
<div id="statistical-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#statistical-analysis" class="anchor"></a>Statistical Analysis</h2>
<p>Once the data have been preprocessed, you can start with the statistical analysis block! <strong>POMA</strong> offers many different statistical methods and possible combinations to compute. However, in this vignette we will comment only some of the most used.</p>
<div id="univariate-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#univariate-analysis" class="anchor"></a>Univariate Analysis</h3>
<p><strong>POMA</strong> allows you to perform all of the most used univariate statistical methods in MS by using only one function! <code>PomaUnivariate</code> wrap 4 different univariate methods (ttest, ANOVA and ANCOVA, Wilcoxon test and Kruskal-Wallis Rank Sum Test) that you can perform changing only the “method” argument.</p>
<div id="t-test" class="section level4">
<h4 class="hasAnchor">
<a href="#t-test" class="anchor"></a>T-test</h4>
<div class="sourceCode" id="cb15"><html><body><pre class="r">PomaUnivariate(pre_processed, method = "ttest") %&gt;% head()
&gt;                    mean_Controls mean_DMD Fold_Change_Ratio Difference_Of_Means
&gt; x1_methylhistidine         -0.40     0.16            -0.400                0.56
&gt; x3_methylhistidine         -0.42     0.19            -0.452                0.61
&gt; alanine                    -0.32     0.11            -0.344                0.43
&gt; arginine                   -0.11     0.06            -0.545                0.17
&gt; asparagine                 -0.35     0.13            -0.371                0.48
&gt; aspartic_acid              -0.25     0.09            -0.360                0.34
&gt;                          pvalue    pvalueAdj
&gt; x1_methylhistidine 9.302122e-08 3.100707e-07
&gt; x3_methylhistidine 7.822367e-03 9.025808e-03
&gt; alanine            6.617797e-04 8.631909e-04
&gt; arginine           4.796275e-01 4.796275e-01
&gt; asparagine         1.279748e-05 2.399528e-05
&gt; aspartic_acid      3.302402e-02 3.538287e-02</pre></body></html></div>
<p>You can also compute a volcano plot using the T-test results. <em>Note that we’re using the non-normalized object to avoid negative values in our data.</em></p>
<div class="sourceCode" id="cb16"><html><body><pre class="r">PomaVolcano(imputed, pval = "adjusted")
&gt; Warning in PomaVolcano(imputed, pval = "adjusted"): adjust argument is empty!
&gt; FDR will be used</pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-16-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
</div>
<div id="wilcoxon-test" class="section level4">
<h4 class="hasAnchor">
<a href="#wilcoxon-test" class="anchor"></a>Wilcoxon Test</h4>
<div class="sourceCode" id="cb17"><html><body><pre class="r">PomaUnivariate(pre_processed, method = "mann") %&gt;% head()
&gt;                    mean_Controls mean_DMD Fold_Change_Ratio Difference_Of_Means
&gt; x1_methylhistidine         -0.40     0.16            -0.400                0.56
&gt; x3_methylhistidine         -0.42     0.19            -0.452                0.61
&gt; alanine                    -0.32     0.11            -0.344                0.43
&gt; arginine                   -0.11     0.06            -0.545                0.17
&gt; asparagine                 -0.35     0.13            -0.371                0.48
&gt; aspartic_acid              -0.25     0.09            -0.360                0.34
&gt;                          pvalue    pvalueAdj
&gt; x1_methylhistidine 6.206369e-05 0.0001432239
&gt; x3_methylhistidine 9.989212e-03 0.0115260136
&gt; alanine            2.152908e-04 0.0004036702
&gt; arginine           1.400885e-01 0.1449191650
&gt; asparagine         2.468467e-04 0.0004356118
&gt; aspartic_acid      5.709276e-03 0.0068511312</pre></body></html></div>
</div>
</div>
<div id="limma" class="section level3">
<h3 class="hasAnchor">
<a href="#limma" class="anchor"></a>Limma</h3>
<p>Other of the wide used statistical methods in many different omics, such as epigenomics or transcriptomics, is <strong>limma</strong> <span class="citation">(Ritchie et al. <a href="#ref-limma">2015</a>)</span>. <strong>POMA</strong> provides an easy use implementation of <em>limma</em> you only have to specify the desired contrast to compute.</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r">PomaLimma(pre_processed, contrast = "Controls-DMD", adjust = "fdr") %&gt;% head()
&gt;                 logFC AveExpr         t      P.Value    adj.P.Val         B
&gt; tryptophan -0.7749207 0.00862 -7.006568 2.691438e-09 8.074315e-08 10.982005
&gt; valine     -0.7009883 0.01268 -6.631334 1.155051e-08 1.732576e-07  9.553920
&gt; ornithine  -0.6327809 0.03366 -6.262549 4.794292e-08 4.794292e-07  8.160589
&gt; isoleucine -0.6058485 0.00438 -5.948984 1.591994e-07 1.193995e-06  6.987716
&gt; lactate    -0.7853613 0.01840 -5.687122 4.296261e-07 2.577757e-06  6.019252
&gt; pyruvate   -0.6244615 0.01208 -5.432256 1.117445e-06 5.135059e-06  5.088552</pre></body></html></div>
</div>
<div id="multivariate-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#multivariate-analysis" class="anchor"></a>Multivariate Analysis</h3>
<p>On the other hand, multivariate analysis implemented in <strong>POMA</strong> is quite similar to the univariate approaches. <code>PomaMultivariate</code> allows users to compute a PCA, PLS-DA or sPLS-DA by changing only the “method” parameter. This function is based on <strong>mixOmics</strong> package <span class="citation">(Rohart et al. <a href="#ref-mixOmics">2017</a>)</span>.</p>
<div id="principal-component-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#principal-component-analysis" class="anchor"></a>Principal Component Analysis</h4>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="no">poma_pca</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/PomaMultivariate.html">PomaMultivariate</a></span>(<span class="no">pre_processed</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">"pca"</span>)</pre></body></html></div>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="no">poma_pca</span>$<span class="no">screeplot</span> +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Scree Plot"</span>)</pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-20-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">poma_pca</span>$<span class="no">scoresplot</span> +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Scores Plot"</span>)</pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-21-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
</div>
<div id="pls-da" class="section level4">
<h4 class="hasAnchor">
<a href="#pls-da" class="anchor"></a>PLS-DA</h4>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="no">poma_plsda</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/PomaMultivariate.html">PomaMultivariate</a></span>(<span class="no">pre_processed</span>, <span class="kw">method</span> <span class="kw">=</span> <span class="st">"plsda"</span>)</pre></body></html></div>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="no">poma_plsda</span>$<span class="no">scoresplot</span> +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Scores Plot"</span>)</pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-23-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="no">poma_plsda</span>$<span class="no">errors_plsda_plot</span> +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">"Error Plot"</span>)</pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-24-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="correlation-analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#correlation-analysis" class="anchor"></a>Correlation Analysis</h3>
<p>Often, correlation analysis is used to explore and discover relationships and patterns within our data. <code>PomaCorr</code> provides a flexible and easy way to do that providing a table with all pairwise coorelations in the data, a correlogram and a correlation graph.</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r">poma_cor &lt;- PomaCorr(pre_processed, label_size = 8, coeff = 0.6)
poma_cor$correlations %&gt;% head()
&gt;           Var1      Var2      corr
&gt; 341 isoleucine   leucine 0.9631456
&gt; 642    leucine    valine 0.9405392
&gt; 836   fumarate    malate 0.9398663
&gt; 641 isoleucine    valine 0.9378129
&gt; 545 asparagine threonine 0.9067625
&gt; 558     serine threonine 0.8931187
poma_cor$corrplot</pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-25-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="no">poma_cor</span>$<span class="no">graph</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-25-2.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<p>Alternatively, if you switch the “corr_type” parameter to “glasso”, this function will compute a <strong>Gaussian Graphical Model</strong> using the <strong>glmnet</strong> package <span class="citation">(Friedman, Hastie, and Tibshirani <a href="#ref-glasso">2019</a>)</span>.</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="../reference/PomaCorr.html">PomaCorr</a></span>(<span class="no">pre_processed</span>, <span class="kw">corr_type</span> <span class="kw">=</span> <span class="st">"glasso"</span>, <span class="kw">coeff</span> <span class="kw">=</span> <span class="fl">0.6</span>)$<span class="no">graph</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-26-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
</div>
<div id="lasso-ridge-and-elasticnet" class="section level3">
<h3 class="hasAnchor">
<a href="#lasso-ridge-and-elasticnet" class="anchor"></a>Lasso, Ridge and Elasticnet</h3>
<p><strong>POMA</strong> also provides a function to perform a Lasso, Ridge and Elasticnet regression for binary outcomes in a very intuitive and easy way. <code>PomaLasso</code> is based on <strong>glmnet</strong> package <span class="citation">(Friedman, Hastie, and Tibshirani <a href="#ref-glmnet">2010</a>)</span>. This function allows you to create a test subset in your data, evaluate the prediction of your models and export the model computed (it could be useful to perform prediction models with MS data). If “ntest” parameter is set to NULL, <code>PomaLasso</code> will use all observations to create the model (useful for feature selection).</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="co"># alpha = 1 for Lasso</span>
<span class="fu"><a href="../reference/PomaLasso.html">PomaLasso</a></span>(<span class="no">pre_processed</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">1</span>)$<span class="no">coefficientPlot</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-27-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="co"># alpha = 0 for Ridge</span>
<span class="fu"><a href="../reference/PomaLasso.html">PomaLasso</a></span>(<span class="no">pre_processed</span>, <span class="kw">alpha</span> <span class="kw">=</span> <span class="fl">0</span>)$<span class="no">coefficientPlot</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-28-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
</div>
<div id="random-forest" class="section level3">
<h3 class="hasAnchor">
<a href="#random-forest" class="anchor"></a>Random Forest</h3>
<p>Finally, the random forest algorithm is also implemented in <strong>POMA</strong>. <code>PomaRandForest</code> uses the <strong>randomForest</strong> package <span class="citation">(Liaw and Wiener <a href="#ref-randomForest">2002</a>)</span> to facilitate the implementation of the algorithm and creates automatically both test and train sets to compute and evaluate the resultant models.</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="no">poma_rf</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/PomaRandForest.html">PomaRandForest</a></span>(<span class="no">pre_processed</span>, <span class="kw">ntest</span> <span class="kw">=</span> <span class="fl">10</span>, <span class="kw">nvar</span> <span class="kw">=</span> <span class="fl">10</span>)
<span class="no">poma_rf</span>$<span class="no">error_tree</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-29-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
<p>Resultant random forest model confusion matrix for <strong>test</strong> set:</p>
<div class="sourceCode" id="cb31"><html><body><pre class="r">poma_rf$confusion_matrix
&gt;          Controls DMD class.error
&gt; Controls        1   1         0.5
&gt; DMD             0   3         0.0</pre></body></html></div>
<p>Gini index plot for the top 10 predictors:</p>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="no">poma_rf</span>$<span class="no">gini_plot</span></pre></body></html></div>
<p><img src="POMA-demo_files/figure-html/unnamed-chunk-31-1.png" width="1458.33333333333" style="display: block; margin: auto;"></p>
</div>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<div class="sourceCode" id="cb33"><html><body><pre class="r">sessionInfo()
&gt; R version 4.0.1 (2020-06-06)
&gt; Platform: x86_64-apple-darwin17.0 (64-bit)
&gt; Running under: macOS Catalina 10.15.4
&gt; 
&gt; Matrix products: default
&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
&gt; 
&gt; locale:
&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
&gt; 
&gt; attached base packages:
&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
&gt; 
&gt; other attached packages:
&gt; [1] ggraph_2.0.3     ggplot2_3.3.2    POMA_0.99.15     BiocStyle_2.16.0
&gt; 
&gt; loaded via a namespace (and not attached):
&gt;   [1] backports_1.1.8       circlize_0.4.10       plyr_1.8.6           
&gt;   [4] igraph_1.2.5          lazyeval_0.2.2        splines_4.0.1        
&gt;   [7] gmp_0.6-0             BiocParallel_1.22.0   digest_0.6.25        
&gt;  [10] foreach_1.5.0         htmltools_0.5.0       viridis_0.5.1        
&gt;  [13] magrittr_1.5          memoise_1.1.0         cluster_2.1.0        
&gt;  [16] doParallel_1.0.15     limma_3.44.3          recipes_0.1.13       
&gt;  [19] ComplexHeatmap_2.4.2  graphlayouts_0.7.0    gower_0.2.2          
&gt;  [22] matrixStats_0.56.0    rARPACK_0.11-0        pkgdown_1.5.1        
&gt;  [25] colorspace_1.4-1      ggrepel_0.8.2         xfun_0.15            
&gt;  [28] dplyr_1.0.0           jsonlite_1.7.0        crayon_1.3.4         
&gt;  [31] impute_1.62.0         survival_3.2-3        iterators_1.0.12     
&gt;  [34] glue_1.4.1            polyclip_1.10-0       gtable_0.3.0         
&gt;  [37] ipred_0.9-9           zlibbioc_1.34.0       GetoptLong_1.0.0     
&gt;  [40] RankProd_3.14.0       shape_1.4.4           Rmpfr_0.8-1          
&gt;  [43] BiocGenerics_0.34.0   scales_1.1.1          vsn_3.56.0           
&gt;  [46] Rcpp_1.0.4.6          mzR_2.22.0            viridisLite_0.3.0    
&gt;  [49] clue_0.3-57           preprocessCore_1.50.0 clisymbols_1.2.0     
&gt;  [52] stats4_4.0.1          lava_1.6.7            prodlim_2019.11.13   
&gt;  [55] glmnet_4.0-2          httr_1.4.1            htmlwidgets_1.5.1    
&gt;  [58] RColorBrewer_1.1-2    ellipsis_0.3.1        pkgconfig_2.0.3      
&gt;  [61] XML_3.99-0.3          farver_2.0.3          nnet_7.3-14          
&gt;  [64] caret_6.0-86          labeling_0.3          tidyselect_1.1.0     
&gt;  [67] rlang_0.4.6           reshape2_1.4.4        ggcorrplot_0.1.3     
&gt;  [70] munsell_0.5.0         tools_4.0.1           generics_0.0.2       
&gt;  [73] broom_0.5.6           evaluate_0.14         stringr_1.4.0        
&gt;  [76] mzID_1.26.0           yaml_2.2.1            ModelMetrics_1.2.2.2 
&gt;  [79] knitr_1.29            fs_1.4.1              tidygraph_1.2.0      
&gt;  [82] randomForest_4.6-14   purrr_0.3.4           ncdf4_1.17           
&gt;  [85] glasso_1.11           nlme_3.1-148          compiler_4.0.1       
&gt;  [88] rstudioapi_0.11       plotly_4.9.2.1        png_0.1-7            
&gt;  [91] affyio_1.58.0         tibble_3.0.1          tweenr_1.0.1         
&gt;  [94] stringi_1.4.6         desc_1.2.0            RSpectra_0.16-0      
&gt;  [97] MSnbase_2.14.2        lattice_0.20-41       ProtGenerics_1.20.0  
&gt; [100] Matrix_1.2-18         permute_0.9-5         vegan_2.5-6          
&gt; [103] vctrs_0.3.1           pillar_1.4.4          lifecycle_0.2.0      
&gt; [106] BiocManager_1.30.10   MALDIquant_1.19.3     GlobalOptions_0.1.2  
&gt; [109] data.table_1.12.8     corpcor_1.6.9         patchwork_1.0.1      
&gt; [112] R6_2.4.1              pcaMethods_1.80.0     affy_1.66.0          
&gt; [115] bookdown_0.20         gridExtra_2.3         IRanges_2.22.2       
&gt; [118] codetools_0.2-16      MASS_7.3-51.6         assertthat_0.2.1     
&gt; [121] rprojroot_1.3-2       rjson_0.2.20          withr_2.2.0          
&gt; [124] S4Vectors_0.26.1      mgcv_1.8-31           parallel_4.0.1       
&gt; [127] mixOmics_6.12.1       grid_4.0.1            rpart_4.1-15         
&gt; [130] timeDate_3043.102     tidyr_1.1.0           class_7.3-17         
&gt; [133] rmarkdown_2.3         ggforce_0.3.2         pROC_1.16.2          
&gt; [136] Biobase_2.48.0        lubridate_1.7.9       ellipse_0.4.2</pre></body></html></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-imputation">
<p>Armitage, Emily Grace, Joanna Godzien, Vanesa Alonso-Herranz, Ángeles López-Gonzálvez, and Coral Barbas. 2015. “Missing Value Imputation Strategies for Metabolomics Data.” <em>Electrophoresis</em> 36 (24). Wiley Online Library: 3050–60.</p>
</div>
<div id="ref-normalization">
<p>Berg, Robert A van den, Huub CJ Hoefsloot, Johan A Westerhuis, Age K Smilde, and Mariët J van der Werf. 2006. “Centering, Scaling, and Transformations: Improving the Biological Information Content of Metabolomics Data.” <em>BMC Genomics</em> 7 (1). Springer: 142.</p>
</div>
<div id="ref-glasso">
<p>Friedman, Jerome, Trevor Hastie, and Rob Tibshirani. 2019. <em>Glasso: Graphical Lasso: Estimation of Gaussian Graphical Models</em>. <a href="https://CRAN.R-project.org/package=glasso">https://CRAN.R-project.org/package=glasso</a>.</p>
</div>
<div id="ref-glmnet">
<p>Friedman, Jerome, Trevor Hastie, and Robert Tibshirani. 2010. “Regularization Paths for Generalized Linear Models via Coordinate Descent.” <em>Journal of Statistical Software</em> 33 (1): 1–22. <a href="http://www.jstatsoft.org/v33/i01/">http://www.jstatsoft.org/v33/i01/</a>.</p>
</div>
<div id="ref-MSnbase">
<p>Gatto, Laurent, and Kathryn Lilley. 2012. “MSnbase - an R/Bioconductor Package for Isobaric Tagged Mass Spectrometry Data Visualization, Processing and Quantitation.” <em>Bioinformatics</em> 28: 288–89.</p>
</div>
<div id="ref-Biobase">
<p>Huber, W., V. J. Carey, R. Gentleman, S. Anders, M. Carlson, B. S. Carvalho, H. C. Bravo, et al. 2015. “Orchestrating High-Throughput Genomic Analysis with Bioconductor.” <em>Nature Methods</em> 12 (2): 115–21. <a href="http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html">http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html</a>.</p>
</div>
<div id="ref-randomForest">
<p>Liaw, Andy, and Matthew Wiener. 2002. “Classification and Regression by randomForest.” <em>R News</em> 2 (3): 18–22. <a href="https://CRAN.R-project.org/doc/Rnews/">https://CRAN.R-project.org/doc/Rnews/</a>.</p>
</div>
<div id="ref-limma">
<p>Ritchie, Matthew E, Belinda Phipson, Di Wu, Yifang Hu, Charity W Law, Wei Shi, and Gordon K Smyth. 2015. “limma Powers Differential Expression Analyses for RNA-Sequencing and Microarray Studies.” <em>Nucleic Acids Research</em> 43 (7): e47. <a href="https://doi.org/10.1093/nar/gkv007">https://doi.org/10.1093/nar/gkv007</a>.</p>
</div>
<div id="ref-mixOmics">
<p>Rohart, Florian, Benoît Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017. “MixOmics: An R Package for ’Omics Feature Selection and Multiple Data Integration.” <em>PLoS Computational Biology</em> 13 (11). Public Library of Science: e1005752. <a href="http://www.mixOmics.org">http://www.mixOmics.org</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Pol Castellano-Escuder.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
